code/src/layouts/header/components/header-search-input/HeaderSearchInput.tsx

```diff
@@ -1,29 +1,103 @@
-import { ChangeEvent, useState } from "react";
+import { skipToken } from "@reduxjs/toolkit/query/react";
+import { ChangeEvent, useRef, useState } from "react";
+import { useIntl } from "react-intl";
 
-import AppInputWithIcon from "@/components/app-input-with-icon/AppInputWithIcon";
+import ClearIcon from "@mui/icons-material/Clear";
+import SearchIcon from "@mui/icons-material/Search";
+
+import SearchInputDropdown from "@/layouts/header/components/header-search-input-dropdown/HeaderSearchInputDropdown";
+
+import AppBox from "@/components/app-box/AppBox";
+import AppIconButton from "@/components/app-icon-button/AppIconButton";
+import AppInputBase from "@/components/app-input-base/AppInputBase";
+
+import { useLocaleContext } from "@/context/i18n/I18nProvider";
+import useDebouncedValue from "@/hooks/use-debounced-value/useDebouncedValue";
+import { useOnClickOutside } from "@/hooks/use-on-click-outside/useOnClickOutside";
+import { useGetUserProductsBySearchQuery } from "@/store/api/productsApi";
+import cn from "@/utils/cn/cn";
+
+import "@/layouts/header/components/header-search-input/HeaderSearchInput.scss";
 
 const HeaderSearchInput = () => {
-  const [searchValue, setSearchValue] = useState("");
+  const [searchQuery, setSearchQuery] = useState("");
+  const [isDropdownOpened, setIsDropdownOpened] = useState(false);
+  const { locale: lang } = useLocaleContext();
+  const searchInputRef = useRef<HTMLInputElement>(null);
```

Comment:
I have a suggestion:

```tsx
const searchInputRef = useRef<HTMLInputElement | null>(null);
```

Reply:
but it's already includes this type

Reply:
It does not include this type i believe. It makes our intentions more clear. We can discuss it if needed

Reply:
I checked it. Refactored according to your suggestion